scijava-ops-engine/src/main/java/org/scijava/ops/engine/impl/DefaultOpEnvironment.java

```diff
@@ -664,10 +661,17 @@ private List<RichOp<?>> resolveOpDependencies(OpInfo info,
 	private OpRequest inferOpRequest(OpDependencyMember<?> dependency,
 		Map<TypeVariable<?>, Type> typeVarAssigns)
 	{
+		final Map<TypeVariable<?>, Type> inferences = new HashMap<>(typeVarAssigns);
+		for (var entry : inferences.entrySet()) {
+			var value = entry.getValue();
+			if (value.equals(Any.class) || value instanceof Any) {
+				inferences.put(entry.getKey(), entry.getKey());
+			}
+		}
 		final Type mappedDependencyType = Types.mapVarToTypes(new Type[] {
-			dependency.getType() }, typeVarAssigns)[0];
+			dependency.getType() }, inferences)[0];
 		final String dependencyName = dependency.getDependencyName();
-		return inferOpRequest(mappedDependencyType, dependencyName, typeVarAssigns);
+		return inferOpRequest(mappedDependencyType, dependencyName, inferences);
```

Comment:
This needs documentation (and maybe also cleanup), if it is indeed necessary.

For whom it may concern, remember that the basic idea is that the `Any` type is used to say "this is going to be some concrete type, but I don't care what type that is", and is used in many situations including function calls without a defined output type, and in simplification - this idea can hurt `OpDependency` parameters.

Consider, for example, the following `copy` Op:
```java
public class CopyOp implements Computers.Arity1<RAI<I>, RAI<O>> {
    @OpDependency(name="engine.copy")
    private final Computers.Arity1<I, O> copier;
    
    @Override
    public void compute(RAI<I> in, RAI<O> out) {...}
} 
```

If I then want to call

```java
var myCopy = ops.op("myCopyOp").input(myImg).apply()
```
I can, using adaptation, match a `Function<RAI<I>, Any>` where the `O` matches to some unknown type. We cannot propagate that `Any` into the dependency, though, as the `copier` might be matched with a `I -> I` copier.

Reply:
Wrote a large comment and added a test that shows the code is necessary